src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/SqlConnectionX.cs

```diff
@@ -85,13 +150,46 @@ public override Task ChangeDatabaseAsync(string databaseName, CancellationToken
         public object Clone()
             => throw new NotImplementedException();
 
+        #region close
+
         /// <inheritdoc/>
         public override void Close()
-            => throw new NotImplementedException();
+            => Close(async: false).GetAwaiter().GetResult();
 
         /// <inheritdoc/>
         public override Task CloseAsync()
-            => throw new NotImplementedException();
+            => Close(async: true);
+
+        internal Task Close(bool async)
+        {
+            //TODO: make thread safe?
+
+            switch (State)
+            {
+                case ConnectionState.Open:
+                case ConnectionState.Executing:
+                case ConnectionState.Fetching:
+                    break;
+                case ConnectionState.Connecting:
+                    //TODO: change this to match current behavior. cancels any pending async open task
+                    throw new InvalidOperationException("Can't close connection while in Connecting state");
+                default:
+                    return Task.CompletedTask;
+            }
+
+            var internalConnection = _connection;
+
+            //TODO: cancel outstanding operations on connection
+
+            //TODO: support async close operation
+            internalConnection?.Close();
+
+            internalConnection = null;
```

Comment:
There is no point to set a local variable to `null`.

Reply:
changed this to a TODO. we'll need to clear the internal connection's reference to the wrapper connection.